### PR TITLE
Improve README

### PR DESCRIPTION
**Pull request by naturalwarren**
Revamps the README as discussed with @behroozkhorashadi and @LukeStClair.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,19 +1,17 @@
 ## RAVE (Runtime Annotation Validation Engine) [![Build Status](https://travis-ci.org/uber-common/rave.svg?branch=master)](https://travis-ci.org/uber-common/rave)
 
-RAVE is a data model validation framework that uses Java annotation processing in order to generate runtime code to validate specific data constraints within your data models.
+RAVE is a shield that prevents invalid data from crashing or causing hard to spot bugs in your Android apps. RAVE uses java annotation processing to leverage the annotations ([Nullness](https://developer.android.com/studio/write/annotations.html#adding-nullness), [Value Constraint](https://developer.android.com/studio/write/annotations.html#value-constraint), [Typedef](https://developer.android.com/studio/write/annotations.html#enum-annotations)) already present in your model classes to increase safety at runtime. Specifically, it ensures models adhere to the set of expectations that are described by their annotations.
```
Comment (LukeStClair) [inline README.md]: maybe "prevents invalid data from entering your app, potentially causing crashes or hard to spot bugs"?

Could go either way.

File: README.md [hunk 2]
```diff
@@ -1,19 +1,17 @@
...
 
 
 ## Motivation
 
-Using Java annotation processing, RAVE provides an efficient mechanism to allow engineers to place restrictions and checks on the data produced and consumed within a client code base. This allows the processor to take additional steps such as generate compiler errors, warnings and even generate source code.
-
-RAVE will leverage the user-provided annotation to provide concise and efficient code to validate model data used in your Android Java code base. Regardless of the transport format, RAVE gives us flexible means to ensure that we are getting the data we expect and continue to expect throughout the data model lifecycle.
+Android apps consume data from a variety of sources (network, disk, etc) that you as an app developer don’t have control over. When external APIs behave badly and return `null` for something that’s supposed to be `@NonNull`, your app can crash. Even when APIs behave well sometimes their corner cases aren’t well documented, are unknown or may change overtime. RAVE ensures the data you receive from these sources adheres to the set of expectations described by the annotations present on your models.
```
Comment (LukeStClair) [inline README.md]: maybe something about continuing to satisfy the contract throughout the whole lifecycle?

Comment (MollyVorwerck) [inline README.md]: Period after etc.

Comment (MollyVorwerck) [inline README.md]: Comma after "Even when APIs behave well,"

Comment (MollyVorwerck) [inline README.md]: well documented --> well-documented

**Pull request merged**